### PR TITLE
Migrate parts of files app away from depecrated Ilogger

### LINKED ISSUE #32127: [Meta]: Port away from ILogger
## Current pull requests
 - [x] https://github.com/nextcloud/server/pull/31609
 - [x] https://github.com/nextcloud/server/pull/31943
 - [x] https://github.com/nextcloud/server/pull/31948
 - [x] https://github.com/nextcloud/server/pull/40508
 - [ ] ...

## Current remaining modules to port in server
 - [x] lib/{private,public}/Files 

## Current remaining shipped app to port
 - [x] federation https://github.com/nextcloud/server/pull/39018
 - [x] encryption https://github.com/nextcloud/server/pull/39065
 - [x] federatedfilesharing #39160 
 - [x] files_sharing https://github.com/nextcloud/server/pull/39169
 - [ ] files_trashbin
 - [x] files_external https://github.com/nextcloud/server/pull/39480
 - [ ] dav
 - [x] weather_status
 - [ ] settings
 - [x] files
 - [ ] workflowengine
 - [x] user_status
 - [x] user_ldap https://github.com/nextcloud/server/pull/40504
 - [x] updatenotification
 - [x] twofactor_backupcodes
 - [x] theming
 - [x] sharebymail

## Current issues

- The background job interface exposes `ILogger` in its public API. We need to deprecate this
- In some places, like the updater. It's helpful to be able to write `ILogger:DEBUG` instead of `2`. We need to look if upstream psr would be interested to add enums for the log levels.

### PR DESCRIPTION
**Pull request by nfebe**
- Migrate some backgrounds jobs, `TransferOwnership` & `DeleteOrphanedItems`
- Migrate `DirectEditingController`

Resolves :  https://github.com/nextcloud/server/issues/32127

### CODE REVIEW

File: apps/files/lib/BackgroundJob/DeleteOrphanedItems.php [hunk 1]
```diff
@@ -27,34 +27,31 @@
...
+	protected LoggerInterface $logger;
 
 	/**
 	 * sets the correct interval for this timed job
 	 */
-	public function __construct(ITimeFactory $time) {
+	public function __construct(
+		ITimeFactory $time,
+
+		protected int $defaultIntervalMin = 60
```
Comment (come-nc) [inline apps/files/lib/BackgroundJob/DeleteOrphanedItems.php:49]: Why do you add this to the constructor parameters?

File: apps/files/lib/BackgroundJob/TransferOwnership.php [hunk 2]
```diff
@@ -87,7 +64,10 @@ protected function run($argument) {
 		$nodes = $userFolder->getById($fileId);
 
 		if (empty($nodes)) {
-			$this->logger->alert('Could not transfer ownership: Node not found');
+			$this->logger->alert(
+				'Could not transfer ownership: Node not found',
+				['app' => 'files'],
```
Comment (come-nc) [inline apps/files/lib/BackgroundJob/TransferOwnership.php:69]: Is that needed? Logger injected through DI is supposed to guess the application itself, I was told. I never checked though.

Comment (ChristophWurst) [reply apps/files/lib/BackgroundJob/TransferOwnership.php:69]: The app containers have a service alias for a logger with the app id preset, yes

Comment (Fenn-CS) [reply apps/files/lib/BackgroundJob/TransferOwnership.php:69]: Thanks @ChristophWurst please can you link the snippet for my benefit?

**Review by come-nc — APPROVED** [review]

File: apps/files/lib/BackgroundJob/DeleteOrphanedItems.php [hunk 3]
```diff
@@ -27,34 +27,28 @@
...
-	public function __construct(ITimeFactory $time) {
+	public function __construct(
+		ITimeFactory $time,
+	) {
 		parent::__construct($time);
 		$this->interval = $this->defaultIntervalMin * 60;
-		$this->connection = \OC::$server->getDatabaseConnection();
-		$this->logger = \OC::$server->getLogger();
+		$this->connection = \OC::$server->get(IDBConnection::class);
+		$this->logger = \OC::$server->get(LoggerInterface::class);
```
Comment (Altahrim) [inline apps/files/lib/BackgroundJob/DeleteOrphanedItems.php:51]: Can't we use dependency injection here?

**Review by Altahrim — APPROVED** [review]

File: apps/files/lib/BackgroundJob/DeleteOrphanedItems.php (hunk 3 above)
Comment (ArtificialOwl) [reply apps/files/lib/BackgroundJob/DeleteOrphanedItems.php:51]: we can and we should

Comment (Fenn-CS) [reply apps/files/lib/BackgroundJob/DeleteOrphanedItems.php]: @Altahrim, kindly review one more time! Made this change!

Thank you.

**Review by Altahrim — APPROVED** [review]

**Comment by solracsf** [comment]
CI Red 🔴

**Comment by come-nc** [comment]
@Fenn-CS You need to update the tests to adapt them to the changes you made to the background job constructor.

**Comment by come-nc** [comment]
@Fenn-CS Please avoid referencing the issue from the commit comment, it is spamming the issue: https://github.com/nextcloud/server/issues/32127

**Comment by Fenn-CS** [comment]
@solracsf fixed related failures, please can you give this one more review so I can manually merge this?

**Review by solracsf — APPROVED** [review]
CI is happy so am I. 🍭

**Pull request merged**